### PR TITLE
added support for miniMACRO5 and bigKNOBv2

### PR DESCRIPTION
**Pull request by LeafCutterLabs**
added support for 2 boards I designed (miniMACRO5 and bigKNOBv2)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by loganmarchione** [comment]
@LeafCutterLabs  - I was going to add bigKNOBv2 myself, but I closed my PR in favor of yours.

You'll need to update the READMEs with some basic info like picture, compilation information, reset information, etc...

You can see an example here:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/planck/readme.md

File: keyboards/leafcutterlabs/bigKNOBv2/bigKNOBv2.c [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include "bigKNOBv2.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/bigKNOBv2/bigKNOBv2.c:5]: This duplicates the default function, so you can remove it here.

https://github.com/qmk/qmk_firmware/blob/ba7f52aaeba878544973952ca3fcce27ab4bb92d/quantum/matrix_common.c#L21

```suggestion
```

File: keyboards/leafcutterlabs/bigKNOBv2/bigKNOBv2.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+#include "quantum.h"
+
+#define LAYOUT_ortho_1x5( \
+    k01, k02, k03, k04, k05 \
+  ) { \
+    { k01, k02, k03, k04, k05 } \
+}
+
+#define LAYOUT LAYOUT_ortho_1x5
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/bigKNOBv2/bigKNOBv2.h:11]: I'd prefer one or the other of `LAYOUT` and `LAYOUT_ortho_1x5` be used here, but not both. So either:


```suggestion
#define LAYOUT( \
    k01, k02, k03, k04, k05 \
  ) { \
    { k01, k02, k03, k04, k05 } \
}
```

or

```suggestion
#define LAYOUT_ortho_1x5( \
    k01, k02, k03, k04, k05 \
  ) { \
    { k01, k02, k03, k04, k05 } \
}
```

Whichever you use will also need to be used in:

- https://github.com/qmk/qmk_firmware/blob/8caef9f0c58c69d22fbc72206b4cfe44564601e4/keyboards/leafcutterlabs/bigKNOBv2/info.json#L8
- https://github.com/qmk/qmk_firmware/blob/8caef9f0c58c69d22fbc72206b4cfe44564601e4/keyboards/leafcutterlabs/bigKNOBv2/keymaps/default/keymap.c#L37
- https://github.com/qmk/qmk_firmware/blob/8caef9f0c58c69d22fbc72206b4cfe44564601e4/keyboards/leafcutterlabs/bigKNOBv2/keymaps/media/keymap.c#L37

File: keyboards/leafcutterlabs/bigKNOBv2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/bigKNOBv2/rules.mk:16]: Some `rules.mk` templating updates...

```suggestion
#   change yes to no to disable
```

File: keyboards/leafcutterlabs/bigKNOBv2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+TAP_DANCE_ENABLE = yes      # Double press does something different
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/bigKNOBv2/rules.mk:34]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI controls
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
UNICODE_ENABLE = yes        # Unicode
TAP_DANCE_ENABLE = yes      # Double press does something different
ENCODER_ENABLE = yes        # Enable rotary encoders
```

Also, you can keep it if you wish, but be advised that `TAP_DANCE_ENABLE = yes` will prevent [QMK Configurator](https://config.qmk.fm/) from compiling *any* firmware for your board.

File: keyboards/leafcutterlabs/minimacro5/minimacro5.c [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#include "minimacro5.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/minimacro5/minimacro5.c:5]: As above.

```suggestion
```

File: keyboards/leafcutterlabs/minimacro5/minimacro5.h [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+#include "quantum.h"
+
+#define LAYOUT_ortho_1x5( \
+    k01, k02, k03, k04, k05\
+  ) { \
+    { k01, k02, k03, k04, k05 } \
+}
+
+#define LAYOUT LAYOUT_ortho_1x5
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/minimacro5/minimacro5.h:11]: As above.

File: keyboards/leafcutterlabs/minimacro5/rules.mk [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/minimacro5/rules.mk:16]:
```suggestion
#   change yes to no to disable
```

File: keyboards/leafcutterlabs/minimacro5/rules.mk [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+TAP_DANCE_ENABLE = yes      # Double press does something different
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/minimacro5/rules.mk:34]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover 
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI controls
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
UNICODE_ENABLE = yes        # Unicode
TAP_DANCE_ENABLE = yes      # Double press does something different
ENCODER_ENABLE = yes        # Enable rotary encoders
```

File: keyboards/leafcutterlabs/bigKNOBv2/config.h [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+#define UNUSED_PINS
+
+/* rotary encoder 1,2,3 closest to usb port is 0*/
+#define ENCODERS_PAD_A { D0}
+#define ENCODERS_PAD_B { D2}
+#define ENCODER_RESOLUTION 2 //default/suggested
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN C7 //D3 - underglow C7 - backlight
```
Comment (noroadsleft) [inline keyboards/leafcutterlabs/bigKNOBv2/config.h:60]: @drashna, this board uses WS2812 LEDs for both backlight and underglow. Do you have any ideas about getting both to work simultaneously?

Comment (drashna) [reply keyboards/leafcutterlabs/bigKNOBv2/config.h:60]: The simple answer is that you can't.  Both use the WS2812 driver, and the same definitions for the driver config.

However, RGB matrix supports "underglow" regions: https://docs.qmk.fm/#/feature_rgb_matrix?id=flags

But that requires that it be part of the same strand/range. 

In fact, the Corne keyboard does precisely that. it runs the underglow first, and then the per key. So you can have either or, if you want.

File: keyboards/leafcutterlabs/bigKNOBv2/readme.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+#bigknobv2
+Macropad Media Controrller with USC-C + RGB underglow + Hotswaps and support for upto 40mm knob
+![](https://imgur.com/8ZBC23k)
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigKNOBv2/readme.md:3]:
```suggestion
# bigKNOBv2

Macropad Media Controller with USB-C + RGB underglow + Hotswap and support for up to 40mm knob

![](https://i.imgur.com/8ZBC23k.jpg)
```

File: keyboards/leafcutterlabs/bigKNOBv2/keymaps/media/keymap.c [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_MEDIA_NEXT_TRACK);
+	unregister_code (KC_MEDIA_NEXT_TRACK);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigKNOBv2/keymaps/media/keymap.c:22]:
```suggestion
    tap_code(KC_MNXT);
```

File: keyboards/leafcutterlabs/bigKNOBv2/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_D);
+	unregister_code (KC_D);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigKNOBv2/keymaps/default/keymap.c:22]:
```suggestion
    tap_code(KC_D);
```

File: keyboards/leafcutterlabs/minimacro5/readme.md [hunk 13]
```diff
@@ -0,0 +1,4 @@
+# miniMACRO5
+QMK-powered macropad based on Arduino Pro Micro. Supports a mix of up to 5 rotary encoders or mechanical switches with optional RGB underglow.
+![](https://github.com/LeafCutterLabs/miniMACRO5/tree/master/photos/miscconfigs.jpg)
+![](https://github.com/LeafCutterLabs/miniMACRO5/tree/master/photos/underglow.jpg)
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/minimacro5/readme.md:4]:
```suggestion
# miniMACRO5

QMK-powered macropad based on Arduino Pro Micro. Supports a mix of up to 5 rotary encoders or mechanical switches with optional RGB underglow.

![](https://github.com/LeafCutterLabs/miniMACRO5/tree/master/photos/miscconfigs.jpg)
![](https://github.com/LeafCutterLabs/miniMACRO5/tree/master/photos/underglow.jpg)
```

File: keyboards/leafcutterlabs/minimacro5/keymaps/media/keymap.c [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_MEDIA_NEXT_TRACK);
+	unregister_code (KC_MEDIA_NEXT_TRACK);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/minimacro5/keymaps/media/keymap.c:46]:
```suggestion
    tap_code(KC_MNXT);
```

File: keyboards/leafcutterlabs/minimacro5/keymaps/fkey/keymap.c [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_F24);
+	unregister_code (KC_F24);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/minimacro5/keymaps/fkey/keymap.c:46]:
```suggestion
    tap_code(KC_F24);
```

File: keyboards/leafcutterlabs/minimacro5/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_E);
+	unregister_code (KC_E);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/minimacro5/keymaps/default/keymap.c:46]:
```suggestion
    tap_code(KC_E);
```

File: keyboards/leafcutterlabs/minimacro5/config.h [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *                  NO_DIODE = switches are directly connected to AVR pins
+ *
+*/
+// #define MATRIX_ROW_PINS { D0, D5 }
+// #define MATRIX_COL_PINS { F1, F0, B0 }
+#define DIRECT_PINS {   \
+    { F4, B6, B2, D7, B4 } \
+}
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/minimacro5/config.h:49]:
```suggestion
#define DIRECT_PINS { { F4, B6, B2, D7, B4 } }
```

**Comment by loganmarchione** [comment]
@LeafCutterLabs  - Just wondered if there was an update on this...

**Pull request closed without merging**